### PR TITLE
Exchange 2016 Compliance Search - integration and playbook

### PR DESCRIPTION
**Pull request by IkaDemisto**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/14122

## Description
The integration can run only on engine without docker as described in the integration tips. 

The integration has the following commands:
1. exchange2016-start-compliance-search
2. exchange2016-get-compliance-search
3. exchange2016-remove-compliance-search
4. exchange2016-purge-compliance-search-results
5. exchange2016-get-compliance-search-purge-status

Functionality and outputs of the commands are identical to `ews-0365` commands that are part of `EWS v2` integration.

Additionally added `Exchange 2016 Search and Delete` playbook that accepts as an input query (KQL) that is used as search conditions for compliance Search.
Structure of the playbook is identical to `Office 365 Search and Delete` playbook.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [x] Documentation https://github.com/demisto/etc/issues/15844
- [x] Code Review

### CODE REVIEW

File: Integrations/integration-Exchange2016_Compliance.yml [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+        return {
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['text'],
+            'Contents': 'Search {} status: {}'.format(search_name, status),
+            'EntryContext': {
+                'EWS.ComplianceSearch(val.Name === obj.Name)': {'Name': search_name, 'Status': status}
+            }
+        }
+
+    def create_ps_file(ps_name, ps_content):
```
Comment (BenJoParadise) [inline Integrations/integration-Exchange2016_Compliance.yml]: Check if Powershell supports reading PS script from string (not from file).
For parallel execution.

Comment (BenJoParadise) [reply Integrations/integration-Exchange2016_Compliance.yml]: Or add guid to files.

File: Integrations/integration-Exchange2016_Compliance.yml [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+            file.write(ps_content)
+        return ps_path
+
+    def delete_ps_file(ps_path):
+        if os.path.exists(ps_path):
+            os.remove(ps_path)
+
+    def start_compliance_search(query):
+        ps_path = create_ps_file('startcs.ps1', STARTCS)
+        output = subprocess.Popen(["powershell.exe", ps_path, "'" + USERNAME + "'", "'" + PASSWORD + "'", "'" + query + "'", "'" + EXCHANGE_FQDN + "'", "$" + str(UNSECURE)], \
```
Comment (BenJoParadise) [inline Integrations/integration-Exchange2016_Compliance.yml]: escape query single quotes

File: Integrations/integration-Exchange2016_Compliance.yml (hunk 1 above)
Comment (IkaDemisto) [reply Integrations/integration-Exchange2016_Compliance.yml]: added guid to files

**Pull request merged**